### PR TITLE
Add wedge to vtk writer

### PR DESCRIPTION
**Pull request by Davknapp**
Added a vtk_writer for wedges. 

Had to change a line in the vtk-writer for quads, because the dimensions did not match of the quad-coords did not match the dimension of triangles and prisms/wedges.

### CODE REVIEW

File: src/mesh/vtk_helper.jl [hunk 1]
```diff
@@ -16,6 +15,11 @@ function n_verts_between(n, from, to)
     return edge_verts
 end
 
+function sort_by_axis(corner_verts)
+    permutation = sortperm([corner_verts[:,i] for i in 1:size(corner_verts)[2]])
```
Comment (jlchan) [inline src/mesh/vtk_helper.jl:19]:
```suggestion
    permutation = sortperm([corner_verts[:, i] for i in 1:size(corner_verts, 2)])
```
Can you also add a comment on what this function does?

File: src/mesh/vtk_helper.jl [hunk 2]
```diff
@@ -115,6 +119,96 @@ function quad_vtk_order(corner_verts, order, dim, skip = false)
...
 end
 
+"""
+    wedge_vtk_order(corner_verts, order, dim)
+
+Compute the coordinates of a VTK_LAGRANGE_wedge of a wedge of order order 
+defined by the coordinates of the vertices given in corner_verts. dim is the
+dimension of the coordinates given. If skip is set to true, the coordinates
+of the vertex- and edge-points aren't computed, which can be used to compute
+points of a VTK_LAGRANGE_WEDGE
```
Comment (jlchan) [inline src/mesh/vtk_helper.jl]:
```suggestion
Compute the coordinates of a `VTK_LAGRANGE_wedge` of a wedge of order `order` 
defined by the coordinates of the vertices given in `corner_verts`. `dim` is the
dimension of the coordinates given. If `skip` is set to `true`, the coordinates
of the vertex- and edge-points aren't computed, which can be used to compute
points of a `VTK_LAGRANGE_WEDGE`
```
The last part "which can be used to compute points of a `VTK_LAGRANGE_WEDGE`" is a little unclear. When would you want to set `skip=true`?

File: src/mesh/vtk_helper.jl [hunk 3]
```diff
@@ -115,6 +119,96 @@ function quad_vtk_order(corner_verts, order, dim, skip = false)
...
+            end
+            coords = hcat(coords, tmp_vec)
+        end
+    end
+    #faces
+    tri_faces = SVector((1,2,3), (4,5,6))
+    quad_faces = SVector((1,2,5,4), (2,3,6,5), (1,3,6,4))
+    #triangular faces
+    for indices in tri_faces
+        tri_nodes = Matrix{Float64}(undef,dim,0)
```
Comment (jlchan) [inline src/mesh/vtk_helper.jl:155]:
```suggestion
        tri_nodes = Matrix{Float64}(undef, dim, 0)
```

File: src/mesh/vtk_helper.jl [hunk 4]
```diff
@@ -115,6 +119,96 @@ function quad_vtk_order(corner_verts, order, dim, skip = false)
...
+        tmp_vec = Vector{Float64}(undef, dim)
+        for j in indices
+            tri_nodes = hcat(tri_nodes,corner_verts[:,j])
```
Comment (jlchan) [inline src/mesh/vtk_helper.jl:158]:
```suggestion
            tri_nodes = hcat(tri_nodes, corner_verts[:, j])
```

File: src/mesh/vtk_helper.jl [hunk 5]
```diff
@@ -115,6 +119,96 @@ function quad_vtk_order(corner_verts, order, dim, skip = false)
...
+        end
+        face_coords = triangle_vtk_order(tri_nodes, order, 3, true)
+        face_coords = sort_by_axis(face_coords)
+        if length(face_coords) > 0
+            for i in range(1,size(face_coords)[2])
```
Comment (jlchan) [inline src/mesh/vtk_helper.jl:163]:
```suggestion
            for i in range(1,size(face_coords, 2))
```

File: src/mesh/vtk_helper.jl [hunk 6]
```diff
@@ -115,6 +119,96 @@ function quad_vtk_order(corner_verts, order, dim, skip = false)
...
+                coords = hcat(coords, tmp_vec)
+            end
+        end
+    end
+    #quadrilateral faces
+    for indices in quad_faces
+        tmp_vec = Vector{Float64}(undef, dim)
+        quad_nodes = Matrix{Float64}(undef, dim,0)
+        for j in indices
+            quad_nodes = hcat(quad_nodes, corner_verts[:,j])
```
Comment (jlchan) [inline src/mesh/vtk_helper.jl:176]:
```suggestion
        quad_nodes = Matrix{Float64}(undef, dim, 0)
        for j in indices
            quad_nodes = hcat(quad_nodes, corner_verts[:, j])
```

File: src/mesh/vtk_helper.jl [hunk 7]
```diff
@@ -115,6 +119,96 @@ function quad_vtk_order(corner_verts, order, dim, skip = false)
...
+                coords = hcat(coords, tmp_vec)
+            end
+        end
+    end
+    #volume
+    e_z = (corner_verts[:,4] - corner_verts[:,1]) ./ order
+    interior_tri_verts = [corner_verts[:,1] corner_verts[:,2] corner_verts[:,3]]
+    face_coords = triangle_vtk_order(interior_tri_verts, order, 3, true)
+    face_coords = sort_by_axis(face_coords)
+    for i in range(1,num_verts_on_edge)
```
Comment (jlchan) [inline src/mesh/vtk_helper.jl:193]:
```suggestion
    for i in range(1, num_verts_on_edge)
```

File: src/mesh/vtk_helper.jl [hunk 8]
```diff
@@ -115,6 +119,96 @@ function quad_vtk_order(corner_verts, order, dim, skip = false)
...
+        tmp_vec = Vector{Float64}(undef, dim)
+        face_coords = face_coords .+ e_z
+        if length(face_coords) > 0
+            for k in range(1,size(face_coords)[2])
```
Comment (jlchan) [inline src/mesh/vtk_helper.jl:197]:
```suggestion
            for k in 1:size(face_coords, 2)
```
For consistency (unless there's a reason you need `range`?)

File: src/mesh/vtk_helper.jl [hunk 9]
```diff
@@ -139,6 +233,28 @@ function vtk_order(::Quad, order)
     return quad_vtk_order(quad_vtk_vertices, order, 2) 
 end
 
+"""
+    vtk_order(elem::Quad, order)
+
+Construct all node-points of a VTK_LAGRANGE_TRIANGLE of order order. The corner-nodes are
```
Comment (jlchan) [inline src/mesh/vtk_helper.jl]:
```suggestion
Construct all node-points of a VTK_LAGRANGE_QUAD of order `order`. The corner-nodes are
```

File: src/mesh/vtk_helper.jl [hunk 10]
```diff
@@ -139,6 +233,28 @@ function vtk_order(::Quad, order)
...
+given by the reference-triangle used by StartUpDG
```
Comment (jlchan) [inline src/mesh/vtk_helper.jl]:
```suggestion
given by the reference-quadrilateral used by StartUpDG
```

File: src/mesh/vtk_helper.jl [hunk 11]
```diff
@@ -139,6 +233,28 @@ function vtk_order(::Quad, order)
...
+function vtk_order(elem::Quad, order)
+    quad_sud_vertices = [-1.0 1.0 1.0 -1.0; -1.0 -1.0 1.0 1.0]
+    return quad_vtk_order(quad_sud_vertices, order, 2) 
+end
+
+"""
+    vtk_order(elem::Wedge, order)
+
+Construct all node-points of a VTK_LAGRANGE_TRIANGLE of order order. The corner-nodes are
+given by the reference-triangle used by StartUpDG
```
Comment (jlchan) [inline src/mesh/vtk_helper.jl]:
```suggestion
Construct all node-points of a VTK_LAGRANGE_WEDGE of order `order`. The corner-nodes are
given by the reference wedge used by StartUpDG
```

**Review by jlchan — CHANGES_REQUESTED** [review]
Looks good! Only a few minor formatting suggestions and then we can merge it.

File: src/mesh/vtk_helper.jl (hunk 2 above)
Comment (Davknapp) [reply src/mesh/vtk_helper.jl]: For wedges never, I removed this part from the documentation

**Review by jlchan — APPROVED** [review]
LGTM - thanks!

**Pull request merged**